plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/split/HudiSplitFactory.java

```diff
@@ -44,57 +42,74 @@ public HudiSplitFactory(
+            }
 
-        long length = fileStatus.length();
-
-        if (length == 0) {
-            return ImmutableList.of(new FileSplit(fileStatus.path(), 0, 0, new String[0]));
-        }
-
-        if (fileStatus.path() instanceof PathWithBootstrapFileStatus) {
-            return ImmutableList.of(new FileSplit(fileStatus.path(), 0, length, (String[]) null));
-        }
+            if (fileStatus.path() instanceof PathWithBootstrapFileStatus) {
+                return ImmutableList.of(new HudiSplit(
+                        fileStatus.path().toString(),
+                        0,
+                        length,
+                        length,
+                        fileStatus.modificationTime(),
+                        ImmutableList.of(),
+                        hudiTableHandle.getRegularPredicates(),
+                        partitionKeys,
+                        hudiSplitWeightProvider.calculateSplitWeight(length)));
+            }
 
-        ImmutableList.Builder<FileSplit> splits = ImmutableList.builder();
-        long splitSize = fileStatus.blockSize();
+            ImmutableList.Builder<HudiSplit> splits = ImmutableList.builder();
+            long splitSize = fileStatus.blockSize();
 
-        long bytesRemaining = length;
-        while (((double) bytesRemaining) / splitSize > SPLIT_SLOP) {
-            splits.add(new FileSplit(fileStatus.path(), length - bytesRemaining, splitSize, (String[]) null));
-            bytesRemaining -= splitSize;
+            long bytesRemaining = length;
+            while (((double) bytesRemaining) / splitSize > SPLIT_SLOP) {
+                splits.add(new HudiSplit(
+                        fileStatus.path().toString(),
+                        length - bytesRemaining,
+                        splitSize,
+                        length,
+                        fileStatus.modificationTime(),
+                        ImmutableList.of(),
+                        hudiTableHandle.getRegularPredicates(),
+                        partitionKeys,
+                        hudiSplitWeightProvider.calculateSplitWeight(splitSize)));
+                bytesRemaining -= splitSize;
+            }
+            if (bytesRemaining != 0) {
+                splits.add(new HudiSplit(
+                        fileStatus.path().toString(),
+                        length - bytesRemaining,
+                        bytesRemaining,
+                        length,
+                        fileStatus.modificationTime(),
+                        ImmutableList.of(),
+                        hudiTableHandle.getRegularPredicates(),
+                        partitionKeys,
+                        hudiSplitWeightProvider.calculateSplitWeight(bytesRemaining)));
+            }
+            return splits.build();
         }
-        if (bytesRemaining != 0) {
-            splits.add(new FileSplit(fileStatus.path(), length - bytesRemaining, bytesRemaining, (String[]) null));
+        catch (IOException e) {
```

Comment:
What throws `IOException`?

Reply:
refactored this to move it to the isDirectory check